src/proof/lfsc/lfsc_node_converter.cpp

```diff
@@ -604,11 +601,43 @@ TypeNode LfscNodeConverter::postConvertType(TypeNode tn)
 
 std::string LfscNodeConverter::getNameForUserName(const std::string& name)
 {
+  std::stringstream ssan;
   std::stringstream ss;
-  ss << "cvc." << name;
+  // We also must sanitize symbols that are not allowed in LFSC identifiers
+  // here. For the sake of generating unique symbols, we record the santization
+  // as a prefix before the "." separating the user name. Thus, e.g.
+  //   cvc.|a b|
+  // is converted to:
+  //   cvc_32@1.ab
+  // where "32" is the unicode value of " ", 32@1 indicates that space was
+  // removed from the user string at position 1.
```

Comment:
Why include an index instead of just replacing the character with its unicode, e.g., `cvc.|a32@b|`?

Reply:
|a32@b| could be the name of the user symbol, that's why the information about santization is pushed before the `.`.

Reply:
Ah I see. I worded my comment above poorly. I was suggesting to add `@` or `.` as a prefix to make sure that they don't clash with user definitions, but this takes care of it then.